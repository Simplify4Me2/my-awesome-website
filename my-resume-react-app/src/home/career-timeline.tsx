import React, { useState } from "react";
import classnames from "classnames";
import {
  Timeline,
  TimelineConnector,
  TimelineContent,
  TimelineDot,
  TimelineItem,
  TimelineOppositeContent,
  TimelineSeparator,
} from "@mui/lab";
import { AccessTime, CorporateFare, CorporateFareTwoTone, School } from "@mui/icons-material";
import { Typography } from "@mui/material";

interface ICareerTimelineProps {
  onTimelineItemSelected: () => void;
}

export function CareerTimeline({ onTimelineItemSelected } : ICareerTimelineProps) {

  const [state, setState] = useState(false);

  const handleLogoClick = (logo: string) => {
    console.log("yoohoo ", logo);
    onTimelineItemSelected();
  };

  return (
    <Timeline position="alternate">
      <TimelineItem>
        <TimelineOppositeContent
          sx={{ m: "auto 0" }}
          align="right"
          variant="body2"
          color="text.secondary"
        >
          2020 - Current
        </TimelineOppositeContent>
        <TimelineSeparator>
          <TimelineConnector />
          <TimelineDot
            color="primary"
            className="hover:cursor-pointer"
            onClick={() => handleLogoClick("Protime")}
          >
            <AccessTime />
          </TimelineDot>
          <TimelineConnector />
        </TimelineSeparator>
        <TimelineContent sx={{ py: "12px", px: 2 }}>
          <Typography variant="h5" component="span">
            Protime
          </Typography>
          <Typography color="text.secondary">
            Much more than time registration and workforce management
          </Typography>
        </TimelineContent>
      </TimelineItem>

      <TimelineItem>
        <TimelineOppositeContent sx={{ py: "12px", px: 2 }}>
          <Typography variant="h5" component="span">
            Cegeka
          </Typography>
          <Typography color="text.secondary">In close cooperation</Typography>
        </TimelineOppositeContent>
        <TimelineSeparator>
          <TimelineConnector />
          <TimelineDot
            color="info"
            className={classnames('hover:cursor-pointer', { 'color' : state ? 'text.secondary' : 'text.primary' })}
            onClick={() => handleLogoClick("Cegeka")}
            onMouseEnter={() => setState(true)}
            onMouseLeave={() => setState(false)}
          >
            {state ? <CorporateFare /> : <CorporateFareTwoTone />}
          </TimelineDot>
          <TimelineConnector />
        </TimelineSeparator>
        <TimelineContent
          sx={{ m: "auto 0" }}
          variant="body2"
          color="text.secondary"
        >
          2017 - 2020
        </TimelineContent>
      </TimelineItem>

      <TimelineItem>
        <TimelineOppositeContent
          sx={{ m: "auto 0" }}
          variant="body2"
          color="text.secondary"
        >
          2015 - 2018
        </TimelineOppositeContent>
        <TimelineSeparator>
          <TimelineConnector />
          <TimelineDot
            color="info"
            className="hover:cursor-pointer"
            onClick={() => handleLogoClick("VDAB")}
          >
            <School />
          </TimelineDot>
          <TimelineConnector />
        </TimelineSeparator>
        <TimelineContent sx={{ py: "12px", px: 2 }}>
          <Typography variant="h5" component="span">
            VDAB / PXL
          </Typography>
          <Typography color="text.secondary">Switch 2 IT</Typography>
        </TimelineContent>
      </TimelineItem>
    </Timeline>
  );
};
