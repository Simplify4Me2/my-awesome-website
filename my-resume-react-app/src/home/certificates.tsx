import React from "react";

export const Certificates: React.FC = () => {
  return (
    <div className="flex m-10">
      <img
        className="h-60 m-4"
        src="https://images.credly.com/images/be8fcaeb-c769-4858-b567-ffaaa73ce8cf/image.png"
        alt="Microsoft Certified Azure Fundamentals"
      />
      <img
        className="h-60 m-4"
        src="https://images.credly.com/images/63316b60-f62d-4e51-aacc-c23cb850089c/azure-developer-associate-600x600.png"
        alt="Microsoft Certified Azure Developer"
      />
      <img
        className="h-60 m-4"
        src="https://images.credly.com/images/c3ab66f8-5d59-4afa-a6c2-0ba30a1989ca/CERT-Expert-DevOps-Engineer-600x600.png"
        alt="Microsoft Certified DevOps Engineer"
      />
    </div>
  );
};
