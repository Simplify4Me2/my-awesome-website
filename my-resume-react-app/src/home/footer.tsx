import { Box } from "@mui/material";

export function Footer() {
    return <Box sx={{ 
        flexGrow: 1, 
        backgroundColor: 'primary.light', 
        color: 'white' ,
        display: 'flex',
        justifyContent: 'center',
        p: 2
        }}>
        &copy; 2023 Vereecke inc. All rights reserved
    </Box>
}