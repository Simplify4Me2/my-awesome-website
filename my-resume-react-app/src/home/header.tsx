import React from "react";
import {
    AppBar,
    Box,
    Button,
    Toolbar,
    Typography,
  } from "@mui/material";
  import { DeveloperMode } from "@mui/icons-material";

export const Header: React.FC = (props) => {
    return <Box sx={{ flexGrow: 1 }}>
    <AppBar position="static">
      <Toolbar>
        <DeveloperMode className="pr-1" />
        <Typography variant="h6" component="div" sx={{ flexGrow: 1 }}>
          Simon Vereecke
        </Typography>
        <Button color="inherit">Contact</Button>
      </Toolbar>
    </AppBar>
  </Box>
}