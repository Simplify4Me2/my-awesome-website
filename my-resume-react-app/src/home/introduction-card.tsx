import React from "react";
import {
  Button,
  Card,
  CardActions,
  CardContent,
  Typography,
} from "@mui/material";

export const IntroductionCard: React.FC = (props) => {
  return (
    <Card className="justify-content m-3 pt-5" sx={{ minWidth: 275 }}>
      <img className="h-72 m-5" src="/index-hero.jpg" alt="azure-themed-hero" />
      <CardContent>
        <Typography sx={{ fontSize: 14 }} color="text.secondary" gutterBottom>
          Hi
        </Typography>
        <Typography variant="h5" component="div">
          I'm Simon
        </Typography>
        <Typography sx={{ mb: 1.5 }} color="text.secondary">
          Full-stack developer
        </Typography>
        <Typography variant="body2">
          My main areas of expertise
          <br />
          include C#, Angular and React.
        </Typography>
        <Typography variant="body2">
          I like to craft solid and scalable frontend products with
          <br />
          great user experience.
        </Typography>
      </CardContent>
      <CardActions>
        <Button size="small">See my work experience</Button>
      </CardActions>
    </Card>
  );
};
