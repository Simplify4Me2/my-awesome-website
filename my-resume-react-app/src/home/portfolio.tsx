import { Card, CardContent, Paper } from "@mui/material";
import React from "react";
import { Link } from "react-router-dom";

interface PortFolioProps {
  elementRef: React.MutableRefObject<HTMLDivElement | null>;
}

export function Portfolio({ elementRef }: PortFolioProps) {
  return (
    <div ref={elementRef} className="m-6">
      <h1>Portfolio</h1>
      <span>
        TODO: Add some small fun projects like{" "}
        <a
          className="underline"
          href="https://michaellackey.com/"
          target="_blank"
          rel="noreferrer"
        >
          this
        </a>
      </span>
      <ul>
        <Card
          sx={{ maxWidth: 150 }}
        >
          <Link to="/tic-tac-toe">
            <img src="/tictactoe_tile.jpeg" />
          </Link>
        </Card>
        <Card
          sx={{ maxWidth: 150 }}
        >
          <Link to="/planning-poker">
            <img src="/planningpoker_tile.jpeg" />
          </Link>
        </Card>
        <li className="underline">
          <Link to="/tic-tac-toe">Tic Tac Toe</Link>
        </li>
        <li className="underline">
          <Link to="/planning-poker">Planning Poker</Link>
        </li>
        <li className="underline">
          <Link to="/waste-snake">Waste snake</Link>
        </li>
        <li className="underline">
          <Link to="/experiment">Experiment</Link>
        </li>
      </ul>
    </div>
  );
}
