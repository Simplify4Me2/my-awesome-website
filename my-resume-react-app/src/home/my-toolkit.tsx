import React, { useState } from "react";
import classNames from "classnames";
import {
  Card,
  Table,
  TableBody,
  TableCell,
  TableContainer,
  TableHead,
  TableRow,
} from "@mui/material";
import { CareerTimeline } from "./career-timeline";

interface MyToolkitProps {
  elementRef: React.MutableRefObject<HTMLDivElement | null>;
}

export function MyToolkit({ elementRef }: MyToolkitProps) {
  const [state, setState] = useState(false);

  return (
    <section ref={elementRef}>
      {/* <Card
        ref={elementRef}
        className="justify-content m-3 pt-5"
        sx={{ minWidth: 275 }}
      > */}
      {/* <span>My Toolkit</span> */}
      <h1 className="mb-2 mt-0 text-5xl font-medium leading-tight text-primary text-center">
        My Toolkit
      </h1>
      <TableContainer>
        <Table>
          <TableHead>
            <TableRow>
              <TableCell>
                <span className="font-bold">Frontend</span>
              </TableCell>
              <TableCell>
                <span className="font-bold">Backend</span>
              </TableCell>
              <TableCell>
                <span className="font-bold">DevOps</span>
              </TableCell>
              <TableCell>
                <span className="font-bold">UX/UI</span>
              </TableCell>
              <TableCell>
                <span className="font-bold">Scrum</span>
              </TableCell>
            </TableRow>
          </TableHead>
          <TableBody>
            <TableRow>
              <TableCell className={classNames({ underline: state })}>
                Angular
              </TableCell>
              <TableCell>C#</TableCell>
              <TableCell>Azure DevOps</TableCell>
              <TableCell>Figma</TableCell>
              <TableCell>Retrospective</TableCell>
            </TableRow>
            <TableRow>
              <TableCell>React</TableCell>
              <TableCell>Domain-Driven Design</TableCell>
              <TableCell>Gitlab</TableCell>
              <TableCell>Prototyping</TableCell>
              <TableCell>Kanban</TableCell>
            </TableRow>
            <TableRow>
              <TableCell>Redux</TableCell>
              <TableCell>Entity Framework</TableCell>
              <TableCell>Octopus</TableCell>
              <TableCell>Material</TableCell>
              <TableCell>Story Mapping</TableCell>
            </TableRow>
            <TableRow>
              <TableCell>HTML</TableCell>
              <TableCell>SQL Server</TableCell>
              <TableCell>Docker</TableCell>
              <TableCell>Invision</TableCell>
              <TableCell>Example Mapping</TableCell>
            </TableRow>
            <TableRow>
              <TableCell>CSS</TableCell>
              <TableCell>CQRS</TableCell>
              <TableCell>ARM</TableCell>
              <TableCell>Design System</TableCell>
              <TableCell>Agile</TableCell>
            </TableRow>
          </TableBody>
        </Table>
      </TableContainer>
      {/* </Card> */}
      <CareerTimeline
        onTimelineItemSelected={() => setState((prev) => !prev)}
      />
    </section>
  );
}
