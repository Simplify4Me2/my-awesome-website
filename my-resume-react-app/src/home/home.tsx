import React, { useRef } from "react";
import { Header } from "../components/header";
import { Container } from "@mui/material";
import { IntroductionCard } from "./introduction-card";
import { Certificates } from "./certificates";
import { Portfolio } from "./portfolio";
import { MyToolkit } from "./my-toolkit";
import { Footer } from "./footer";

export function Home() {
  const myToolkitRef = useRef<HTMLDivElement | null>(null);
  const myPortfolioRef = useRef<HTMLDivElement | null>(null);

  function scrollIntoView(ref: React.MutableRefObject<HTMLElement | null>) {
    const headerHeight = "88px";
    if (ref && ref.current) {
      ref.current.style.scrollMargin = headerHeight;
      ref.current.scrollIntoView({ behavior: "smooth" });
    }
  }

  return (
    <>
      <Header />
      <Container>
        <Container className="relative h-screen">
          <IntroductionCard
            onMyPortfolioButtonClicked={() => scrollIntoView(myPortfolioRef)}
            onMyToolKitButtonClicked={() => scrollIntoView(myToolkitRef)}
          />
          <Certificates />
        </Container>
        <MyToolkit elementRef={myToolkitRef} />
        <Portfolio elementRef={myPortfolioRef} />
      </Container>
      <Footer />
    </>
  );
};
