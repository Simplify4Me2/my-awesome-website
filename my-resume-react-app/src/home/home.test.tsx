import { render } from "@testing-library/react";

import { Home } from "./home";

jest.mock("../components/header");
jest.mock("./my-toolkit", () => ({
  MyToolkit: (props: object) => `Portfolio ${JSON.stringify(props)}`,
}));
jest.mock("./introduction-card");
jest.mock("./certificates");
jest.mock("./portfolio", () => ({
    Portfolio: (props: object) => `Portfolio ${JSON.stringify(props)}`,
}));
jest.mock("./footer");

describe("home", () => {
  it("renders", () => {
    const { container } = render(<Home />);

    expect(container).toMatchSnapshot();
  });
});
