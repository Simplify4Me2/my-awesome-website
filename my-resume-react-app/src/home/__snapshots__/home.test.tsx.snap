// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`home renders 1`] = `
<div>
  <div
    class="flex flex-col items-center"
  />
</div>
`;
